sequencer/src/persistence/sql.rs

```diff
@@ -2067,6 +2067,42 @@ impl Provider<SeqTypes, LeafRequest<SeqTypes>> for Persistence {
     }
 }
 
+#[async_trait]
+impl Provider<SeqTypes, StateCertRequest> for Persistence {
+    #[tracing::instrument(skip(self))]
+    async fn fetch(&self, req: StateCertRequest) -> Option<StateCertQueryData<SeqTypes>> {
+        let mut tx = match self.db.read().await {
+            Ok(tx) => tx,
+            Err(err) => {
+                tracing::warn!("could not open transaction: {err:#}");
+                return None;
+            },
+        };
+
+        match query_as::<(Vec<u8>,)>("SELECT state_cert FROM state_cert WHERE epoch = $1 LIMIT 1")
```

Comment:
This is fetching from the same table as `get_state_cert` in the `SqlStorage`? If so, I think it's pointless. The other impls in this file are attempting to help catchup missing data from the main query service storage by looking at temporary consensus storage (different tables). This is just replicating the query that the query service already makes so it doesn't add anything.

Reply:
Got it. The other place that might contain this information is the next quorum proposal.